servers/src/grin/seed.rs

```diff
@@ -297,14 +305,29 @@ fn listen_for_addrs(
 	p2p: Arc<p2p::Server>,
 	capab: p2p::Capabilities,
 	rx: &mpsc::Receiver<SocketAddr>,
+	connecting_history: &mut Vec<(SocketAddr, DateTime<Utc>)>,
 ) {
 	if peers.peer_count() >= p2p.config.peer_max_count() {
 		// clean the rx messages to avoid accumulating
 		for _ in rx.try_iter() {}
 		return;
 	}
 
+	let now = Utc::now();
 	for addr in rx.try_iter() {
+		// ignore the duplicate connecting to same peer within 10 seconds
```

Comment:
From this section and the code below, looks like a HashMap would make all this a lot easier?

Reply:
Vector is easy to `truncate`, find a position and call one function.
hashmap is easy to use but hard to truncate. Is my understanding correct ?

Pls let me try to refactor with hashmap and compare it with this vector implementation.